src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/ActionModel.cs

```diff
@@ -93,6 +93,18 @@ public ActionModel(ActionModel other)
 
         public IList<ParameterModel> Parameters { get; }
 
+        /// <summary>
+        /// Gets or sets an <see cref="IOutboundParameterTransformer"/> that will be used to transform 
+        /// built-in route parameters such as <c>action</c>, <c>controller</c>, and <c>area</c> as well as
+        /// additional parameters specified by <see cref="RouteValues"/> into static segments in the route template.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This feature only applies when using endpoint routing.
+        /// </para>
+        /// </remarks>
+        public IOutboundParameterTransformer RouteParameterTransformer { get; set; }
```

Comment:
@JamesNK - do you remember why we didn't just do this? Given how much this feature has resonated with users it seems warranted.

Reply:
I think you didn't like it on the model. Since only the convention and internal code accessed it via properties, properties seemed like a low friction way to add the feature.

Reply:
Add it 👍